### PR TITLE
Add Fluentd > cloudwatch log forwarder

### PR DESCRIPTION
**Pull request by kenden**
Picked up/Updated fluentd-cloudwatch changes from PR https://github.com/kubernetes/charts/pull/211

Now using image from https://github.com/fluent/fluentd-kubernetes-daemonset

The AWS keys are added as secrets from files, using Files.Glob (from PR
comment https://github.com/kubernetes/charts/pull/211#issuecomment-286662756

### CODE REVIEW

**Comment by so0k** [comment]
I've been using a private helm chart while testing out several logging targets (logzio,loggly,logentries,...) and found that I still needed the config files in a configmap 😢 

Even though the images are slim (each tag only has the minimum required to target it's logging target), config often needed to be customised further through config files.

**Comment by seanknox** [comment]
@k8s-bot ok to test

**Comment by kenden** [comment]
@so0k I re-added a configMap. It has the same config than what is inside the docker image 
fluent/fluentd-kubernetes-daemonset:v0.12-cloudwatch

**Comment by unguiculus** [comment]
ping @mgoodness

File: incubator/fluentd-cloudwatch/templates/configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "fullname" . }}
```
Comment (unguiculus) [inline incubator/fluentd-cloudwatch/templates/configmap.yaml]: `app: {{ template "name" . }}`

File: incubator/fluentd-cloudwatch/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+version: 0.1.0
+description: A Fluentd CloudWatch Helm chart for Kubernetes.
+keywords:
+- fluentd
+- cloudwatch
+- logging
+sources:
+- https://github.com/kubernetes/charts
+- https://github.com/fluent/fluentd-kubernetes-daemonset
+maintainers:
```
Comment (unguiculus) [inline incubator/fluentd-cloudwatch/Chart.yaml:13]: Use Github usernames.

File: incubator/fluentd-cloudwatch/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+name: fluentd-cloudwatch
+version: 0.1.0
```
Comment (unguiculus) [inline incubator/fluentd-cloudwatch/Chart.yaml]: Add `appVersion`

Comment (unguiculus) [reply incubator/fluentd-cloudwatch/Chart.yaml]: Is there an `icon` you could add?

**Review by unguiculus — CHANGES_REQUESTED** [review]
We've started namespacing templates. Please apply this to your PR as well. See #1785.

**Comment by kenden** [comment]
@unguiculus I updated the Chart as requested

File: incubator/fluentd-cloudwatch/templates/configmap.yaml (hunk 1 above)
Comment (kenden) [reply incubator/fluentd-cloudwatch/templates/configmap.yaml]: done

File: incubator/fluentd-cloudwatch/Chart.yaml (hunk 3 above)
Comment (kenden) [reply incubator/fluentd-cloudwatch/Chart.yaml]: done and done

File: incubator/fluentd-cloudwatch/Chart.yaml (hunk 2 above)
Comment (kenden) [reply incubator/fluentd-cloudwatch/Chart.yaml:13]: done

File: incubator/fluentd-cloudwatch/templates/configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "fluentd-cloudwatch.fullname" . }}
+  labels:
+    app: {{ template "fluentd-cloudwatch.fullname" . }}
```
Comment (unguiculus) [inline incubator/fluentd-cloudwatch/templates/configmap.yaml]: `app: {{ template "fluentd-cloudwatch.name" . }}`

**Review by unguiculus — CHANGES_REQUESTED** [review]
Sorry for the delay. Almost there.

**Comment by kenden** [comment]
@unguiculus Updated

**Review by unguiculus — APPROVED** [review]

**Comment by unguiculus** [comment]
Sorry this took so long in the end.

**Pull request merged**